odbc/src/api/data.rs

```diff
@@ -1141,6 +1253,237 @@ mod unit {
         );
     }
 
+    mod conversion {
+        use std::{collections::HashMap, f64::consts::PI};
+
+        use crate::{api::data::IntoCData, map};
+        use bson::Bson;
+        use constants::{FRACTIONAL_TRUNCATION, INTEGRAL_TRUNCATION, INVALID_CHARACTER_VALUE};
+
+        macro_rules! test_conversion_ok {
+            (input = $input:expr, method = $method:tt, expected = $expected:expr, info = $info:expr) => {
+                let actual = $input.$method();
+                assert!(actual.is_ok(), "{:?}", actual);
+                match actual {
+                    Ok(r) => {
+                        assert_eq!($expected, r.0, "expected {}, got {}", $expected, r.0);
+                        if r.1.is_some() {
+                            let odbc_info = r.1.unwrap().get_sql_state().to_string();
+                            assert_eq!(
+                                $info, odbc_info,
+                                "expected success with info {}, got {}",
+                                $info, odbc_info
+                            );
+                        }
+                    }
+                    Err(e) => unreachable!("should not have had an err {:?}", e),
+                }
+            };
+        }
+        macro_rules! test_conversion_err {
+            (input = $input:expr, method = $method:tt, expected = $expected:expr, info = $info:expr) => {
+                let actual = $input.$method();
+                assert!(actual.is_err(), "{:?}", actual);
+                match actual {
+                    Err(e) => {
+                        assert_eq!(
+                            $info,
+                            e.get_sql_state(),
+                            "expected {}, got {}",
+                            $info,
+                            e.get_sql_state()
+                        );
+                    }
+                    Ok(r) => unreachable!("should not have had an Ok {:?}", r),
+                }
+            };
+        }
+
+        macro_rules! test_it {
+            ($bson:expr,$v:expr) => {
+                $v.iter().for_each(
+                    |(method, expected, test, info)| match (method, test, info) {
+                        (&"i64", Ok(()), info) => {
```

Comment:
I think the `&` here is redundant since "i64" is already a &str. Kind of surprised that clippy didn't complain, so maybe I'm wrong :D